packages/internal/src/utils.js

```diff
@@ -381,3 +381,37 @@ export const fsStreamReady = stream =>
     stream.on('ready', onReady);
     stream.on('error', onError);
   });
+
+/**
+ * @template T
+ * @typedef {object} ListAccumulator
+ * @property {(value: T) => void} push
+ * @property {() => void} clear
+ * @property {() => T[]} snapshot
+ * @property {() => T[]} takeSnapshot
+ */
+
+/**
+ * @template {any} T
+ * @param {(x: T) => T} coerce
```

Comment:
can't `x` be anything and the function will return `T` or throw?
```suggestion
 * @param {(x: unknown) => T} coerce
```

Reply:
turns out makeListAccumulator is vestigial from when I was using it within (what is now called) `atomicRearrange`. I no longer am, and neither is anything else, so I just deleted it from the PR.

Gone